### DISCUSSION TITLE
**3D Printing with Anycubic Mega S**

### DISCUSSION THREAD

**Discussion by Smaragt** [community/community#52931] [Programming Help]
Hello and sorry 

First of all I have to say that I don't speak English and so I have to rely on the help of Deepl and Co. And what is even worse from programming I also have no idea so a total novice.
I have bought an Anycubic Mega S used with it printed everything so far well. Suddenly came a message "T0 sensor abnormal" after research came out that on the board (Trigorilla 0.0.2) something is broken. I got me a new board and there it went.
The components to my printer :
Trigorilla 0.0.2 Marlin 2.0.5.4 Marlin-AI3M
TMC 2208 driver
BMG extruder direct
Z axis optical limit switch
I can use Ponterface to read the list at startup but don't know which version of Marlin with which features to put on the new board.
I have read the beginner help and also how the name is composed but somehow there is always something missing.  
From the example 2 is 
MEGA_S_DGUS_TMC_BLT_10_1.2.0.hex would result for me MEGA_S_1.0_TMC_???.hex 
1.0 = Display
TMC = Driver TMC2208
? = Board 0.0.2
? = Direct extruder 
? = optical limit switch Z axis

Furthermore I have read that the connections for the stepper motors must be changed on the software side. With the PID settings I can not really do anything. My idea was if it is possible from the board to clone the firmware and to play on the new but I was taught that does not work. Oh yes do you need extra data for the drivers ?
I just don't want to break the whole thing irretrievably so my request is can someone help me put it all together and understand it ?

Thanks already

Translated with www.DeepL.com/Translator (free version)

**Comment by cyrusjameskhan** [comment]
I'll try to help you and break down the process step by step.

First, let's clarify the firmware naming convention:

MEGA_S - This refers to the Anycubic Mega S printer model.
1.0 - This refers to the display version.
TMC - This refers to the TMC2208 stepper motor drivers.
??? - These will be additional features, such as board version, direct extruder, and the optical limit switch for the Z-axis.
Before we proceed, you need to determine if you have a BLTouch sensor installed on your printer. If you do, include "BLT" in the firmware name. If not, leave it out.

Now, let's create a firmware name for your printer configuration:

MEGA_S_1.0_TMC_DE_OLS_2.0.5.4.hex

DE - Direct Extruder (BMG direct extruder)
OLS - Optical Limit Switch (Z-axis optical limit switch)
2.0.5.4 - Marlin firmware version (2.0.5.4 in your case)
Next, you need to configure the Marlin firmware for your specific printer setup:

Download Marlin firmware 2.0.5.4 from the official Marlin GitHub repository: https://github.com/MarlinFirmware/Marlin

Download and install Visual Studio Code (VSCode) and the PlatformIO extension.

Open the Marlin folder you downloaded in VSCode.

Locate and open the 'Configuration.h' and 'Configuration_adv.h' files in the Marlin folder.

Modify the necessary settings according to your printer configuration (TMC2208 drivers, BMG direct extruder, and optical limit switch). 
You can find guidance on specific settings in the Marlin documentation: https://marlinfw.org/docs/configuration/configuration.html
Ensure the correct board version is set in the 'Configuration.h' file. For the Trigorilla 0.0.2 board, you should have the following (arduino) line:

#define MOTHERBOARD BOARD_TRIGORILLA_14

Save the changes and build the firmware using the PlatformIO extension in VSCode.
After you have compiled the firmware, you can upload the firmware to your new Trigorilla 0.0.2 board. Use the following resources to guide you through the firmware upload process:

Flashing guide: https://marlinfw.org/docs/basics/install.html
Regarding the connections for the stepper motors and the PID settings, it's possible that you may need to make changes in the 'Configuration.h' and 'Configuration_adv.h' files, but without knowing the specifics of your printer setup, it's hard to provide exact guidance.

If you're unsure about any of these steps or need more help, I recommend seeking assistance from the Anycubic and Marlin community forums or Facebook groups, as there are many experienced users who can provide guidance tailored to your specific situation.
Good luck with your Anycubic Mega S!

**Comment by Smaragt** [comment]
Hello Cyrusjameskhan

First of all, thank you very much for your hard work.
If you want, we can go to you, I'm not sensitive.

I have to mention something about the components
Firmware on the board Marlin 2.0.5.4 Marlin-AI3M
Trigorilla 0.0.2
TMC 2208 driver
BMG extruder direct
Z axis 2 optical limit switch
New to BL Touch

I downloaded "Marlin-2-0-x-Anycubic-i3-MEGA-S-master" as firmware and "MEGA_S_TMC_BLT_10_v1.5.0-b3.hex" as .hex file from knutwurst
Visual Studio Code and PlatformIO are there too.
I translated the Marlin documentation you mentioned yesterday and first have to see if and how I can cope with it.
What information would be needed to create precise instructions?
I rummaged around in Visual Studio and looked at the "Configuration.h" and "Configuration_adv.h" but they contain all the Anycubic printers and somehow I get lost thinking about what I need and what I don't. I don't understand the structure of 4000 lines of text and I may only need a quarter of that, but what does it have to look like.
I'll translate the accompanying text, maybe I'll understand it then.

Thank you again you helped me a lot
Best regards Henry
